hbogolib/handler.py

```diff
@@ -189,17 +270,87 @@ def post_to_hbogo(self, url, headers, data, response_format='json'):
             xbmcgui.Dialog().ok(self.LB_ERROR, self.language(30004))
             return False
 
-    def get_from_hbogo(self, url, response_format='json'):
+    def get_from_cache(self, url_hash):
+        cur = self.db.cursor()
+        cur.execute("SELECT url_hash FROM request_cache_exclude WHERE url_hash=?", (url_hash,), )
+        if cur.fetchone() is not None:
+            self.log("URL IN CACHE EXCLUDE LIST")
+            return False
+        cur.execute("DELETE FROM request_cache WHERE last_update <= datetime('now','-1 day');")
+        self.db.commit()
+        cur.execute("SELECT request_data FROM request_cache WHERE url_hash=?", (url_hash,),)
+        result = cur.fetchone()
+        if result is not None:
+            self.log("URL FOUND IN CACHE")
+            return result[0]
+        return None
+
+    def cache(self, url_hash, data):
+        cur = self.db.cursor()
+        cur.execute("INSERT INTO request_cache(url_hash, request_data, last_update) VALUES (?, ?, datetime('now'))", (url_hash, data), )
+        self.db.commit()
+
+    def exclude_url_from_cache(self, url):
+        try:
+            cur = self.db.cursor()
+            cur.execute("INSERT INTO request_cache_exclude(url_hash) VALUES (?)", (Util.hash225_string(url), ), )
+            self.db.commit()
+        except Exception:
+            self.log("Exclude from cache WARNING: " + traceback.format_exc())
+
+    def clear_request_cache(self):
+        cur = self.db.cursor()
+        cur.execute("DELETE FROM request_cache;")
+        self.db.commit()
+        xbmcgui.Dialog().notification(self.language(30809), self.LB_SUCESS, self.get_resource("icon.png"))
+
+    def clear_image_cache(self):
+        pass
+
+    def img_cache(self, url):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4949643)